### DISCUSSION TITLE
HTTP Interceptor

### DISCUSSION THREAD

**Discussion by notnullonline** [leptos-rs/leptos#2504] [General]
Is there something similar to Agular Http Interceptor? 
I need it in the auth scenario so I can check the presence of the JWT token, etc...
I found <ProtectedRoute... /> component but it is a bit cumbersome to use...

**Comment by gbj** [comment]
Some more details would be helpful. I am not sure I understand how HttpInterceptor and ProtectedRoute are related. Are you talking about checking something on the server or on the client or both? In server functions or in a component? etc.

**Reply by notnullonline** [reply]
Ok, I am building a pure SPA (CSR) application...

As my frontend experience is based on an Angular background I was wondering if there is something similar to [HTTP Interceptor](https://angular.io/guide/http-interceptor-use-cases#set-default-headers) in the Leptos framework. 

I used to use HTTP Interceptor as a central place to deal with all kinds of inbound/outbound HTTP requests, for example:

- check if JWT is about to expire and prevent issuing the request (outbound requests)
- the presence of some possible header values (inbound requests)
- determine and act according to the error returned from the server (inbound requests)
- etc...

Of course, all these can be achieved in some other way but I am curious if there is something similar in the Leptos framework...?

Thanks

**Reply by gbj** [reply]
The answer is no, there is not: the scope of things included in Leptos is somewhere smaller than Angular and larger than React, so it does not, for example, include an HTTP client at all. You should look at `gloo_net` (lighter-weight but fewer features) or `reqwest` (heavier in binary size but more features) as crates that can help you with HTTP requests in the browser.

**Reply by notnullonline** [reply]
I was already on the trail of those two libraries...

Thank you very much! Cheers!
